app/models/user.php

```diff
@@ -0,0 +1,47 @@
+<?php
+class User extends AppModel
+{
+    public function register(RegisterInfo $register_info)
+    {
+        $register_info->validate();
+        if ($register_info->hasError()) {
```

Comment:
error cannot be catch po since I have to check if password will match and if the username exist in the database.